torch/nn/modules/module.py

```diff
@@ -462,7 +462,8 @@ def register_backward_hook(self, hook):
         module has multiple inputs or outputs. The hook should not modify its
         arguments, but it can optionally return a new gradient with respect to
         input that will be used in place of :attr:`grad_input` in subsequent
-        computations.
+        computations. :attr:`grad_input` will only correspond to the inputs given
```

Comment:
I would make this a note, like: "Note: the grad_input tuple only..."

But also, can we be more specific about what grad_input does/doesn't contain? This note seems to refer to an implementation detail, and not to our user-facing contract?

Reply:
I didn't want to elaborate to much here as these are broken anyway and grad_input usually contains random stuff :)

Reply:
Gasp!

Reply:
This is not the function you're looking for !